artifact/image/layerscanning/image/file_node_test.go

```diff
@@ -227,8 +227,8 @@ func TestRead(t *testing.T) {
 			want: "foo",
 		},
 		{
-			name:    "non-existent file",
```

Comment:
Why is this a "forbidden pattern"? AFAIK both non-existent and nonexistent are valid spellings.

Reply:
There's a general rule in [US English](https://www.thesaurus.com/browse/nonexistent#:~:text=Both%20nonexistent%20and%20non%2Dexistent,word%20becomes%20more%20commonly%20used.):
> As with many hyphenated terms, the hyphen is often dropped when the word becomes more commonly used.

See also [Stack Exchange](https://english.stackexchange.com/questions/237113/nonexistent-non-existent-or-non-existent/237129#237129) which references [Merriam Webster](https://www.merriam-webster.com/dictionary/nonexistent).

If this project isn't using US English, I'm happy to drop this commit.